http-client-tls/Network/HTTP/Client/TLS.hs

```diff
@@ -122,7 +134,9 @@ convertConnection conn = makeConnection
 
 -- | Evil global manager, to make life easier for the common use case
 globalManager :: IORef Manager
-globalManager = unsafePerformIO (newManager tlsManagerSettings >>= newIORef)
+globalManager = unsafePerformIO $ do
+    man <- newManager $ mkManagerSettingsContext (Just globalContext) def Nothing
```

Comment:
There _is_ a tricky reason: my lack of clarity. I'm not sure if _all_ of the changes I applied here should go through. I was originally planning on just modifying `globalManager` and not implementing the `globalContext`, and then I took it a step further and defaulted to everyone sharing a `globalContext`. Once we're sure that we can use `globalContext`, I'd revert this hunk. Otherwise, I'd get rid of `globalContext` and inline the init call to here.